integration/firestore/gulpfile.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const gulp = require('gulp');
+const replace = require('gulp-replace');
+const { resolve } = require('path');
+const webpackStream = require('webpack-stream');
+const webpack = require('webpack');
+
+function compileIntTests() {
+  const config = require('../../config/webpack.test');
+  return gulp
+    .src(
+      resolve(__dirname, '../../packages/firestore/test/integration/**/*.ts')
+    )
+    .pipe(
+      replace(
+        /import firebase from ('|")[^\1]+firebase_export\1;?/,
```

Comment:
Can you comment this regex?  I am having a real hard time parsing it...

FWIW, the goal with firebase_export.ts was to have a single file we could swap out in the integration tests.  If you're doing this swapping by running a regex over the entire test suite, then I don't think we need firebase_export.ts anymore...  Though, TBH, using firebase_export.ts seems safer to me (if you have a bug in your regex, you might just accidentally run against sources, right)?

If you keep firebase_export.ts can you update the comment in it?  I think it says the file is overwritten by the integration test, which is no longer true...

Reply:
Sure I'll add a comment explaining it.

So I'm actually in favor of keeping the `firebase_export.ts` as it gives me a predictable path to replace. If there is a bug in the regex, you are right, it does hit the original source.

I'll update the comment to explain what it is being used for. I'll also refactor the regex a tad to be a little more lenient on whitespace and things.